client/state/reader/site-blocks/reducer.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_SITE_BLOCK_REQUEST_SUCCESS,
+	READER_SITE_UNBLOCK_REQUEST_SUCCESS,
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+
+/**
+ * Tracks all known block statuses, indexed by site ID.
+ *
+ * @param  {Array} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Array}        Updated state
+ */
+export function items( state = {}, action ) {
+	switch ( action.type ) {
+		case READER_SITE_BLOCK_REQUEST_SUCCESS:
+			return {
+				...state,
+				[ action.siteId ]: action.data.success
```

Comment:
what would be a potential reason for a failure? Does it cleanly fall into 400s territory or does it make most sense to keep it a 200 and with the success parameter in the response body as it is now?